src/common/servers/serverManager.ts

```diff
@@ -0,0 +1,426 @@
+        const server = new MattermostServer(team, isPredefined);
+        this.servers.set(server.id, server);
+
+        log.debug('initialized server', server.id);
+
+        const tabOrder: string[] = [];
+        team.tabs.sort((a, b) => a.order - b.order).forEach((tab) => {
+            const tabView = this.getTabView(server, tab.name, tab.isOpen);
+            log.debug('initialized tab', tabView.id);
+
+            this.tabs.set(tabView.id, tabView);
+            tabOrder.push(tabView.id);
+        });
+        this.tabOrder.set(server.id, tabOrder);
+        if (typeof team.lastActiveTab !== 'undefined') {
+            this.lastActiveTab.set(server.id, tabOrder[team.lastActiveTab]);
+        }
+        return server.id;
+    }
+
+    private getFirstOpenTabForServer = (serverId: string) => {
+        const tabOrder = this.tabOrder.get(serverId);
+        if (!tabOrder) {
+            throw new Error(`Cannot find tabs for server id ${serverId}`);
+        }
+        const openTabs = this.getOrderedTabsForServer(serverId).filter((tab) => tab.isOpen);
+        const firstTab = openTabs[0];
+        if (!firstTab) {
+            throw new Error(`No tabs open for server id ${serverId}`);
+        }
+        return firstTab;
+    }
+
+    private persistServers = async (lastActiveTeam?: number) => {
+        this.emit(SERVERS_UPDATE);
+
+        const localServers = [...this.servers.values()].
+            filter((server) => !server.isPredefined).
+            map((server) => this.toConfigTeam(server));
+        await Config.setServers(localServers, lastActiveTeam);
+    }
+
+    private getLastActiveTab = (serverId: string) => {
+        let lastActiveTab: number | undefined;
+        if (this.lastActiveTab.has(serverId)) {
+            const index = this.tabOrder.get(serverId)?.indexOf(this.lastActiveTab.get(serverId)!);
+            if (typeof index !== 'undefined' && index >= 0) {
+                lastActiveTab = index;
+            }
+        }
+        return lastActiveTab;
+    }
+
+    private toConfigTeam = (server: MattermostServer): ConfigTeam => {
+        return {
+            name: server.name,
+            url: `${server.url}`,
+            order: this.serverOrder.indexOf(server.id),
+            lastActiveTab: this.getLastActiveTab(server.id),
+            tabs: this.tabOrder.get(server.id)?.reduce((tabs, tabId, index) => {
+                const tab = this.tabs.get(tabId);
+                if (!tab) {
+                    return tabs;
+                }
+                tabs.push({
+                    name: tab?.type,
+                    order: index,
+                    isOpen: tab.isOpen,
+                });
+                return tabs;
+            }, [] as ConfigTab[]) ?? [],
+        };
+    }
+
+    private getTabView = (srv: MattermostServer, tabName: string, isOpen?: boolean) => {
+        switch (tabName) {
+        case TAB_MESSAGING:
+            return new MessagingTabView(srv, isOpen);
```

Comment:
ok, I need an explanation as to why we do create a new object every time we call `getTabView`